apps/condo/domains/news/schema/ExportNewsRecipientsService.js

```diff
@@ -0,0 +1,205 @@
+        ...acc,
+        ...getUnitsFromSection(section),
+    ]), []) || []
+)
+
+const getUnitsFromSection = (section) => section.floors.flatMap(floor => floor.units.map(unit => ({ unitName: unit.label, unitType: unit.unitType })))
+
+const buildExportFile = async ({ rows, locale }) => {
+    const YesMessage = i18n('Yes', { locale })
+    const NoMessage = i18n('No', { locale })
+    const HeaderMessage =  i18n('excelExport.sheetNames.newsRecipients', { locale })
+
+    const processedRows = rows.reduce((acc, row) => {
+        row.hasResident = row.hasResident ? YesMessage : NoMessage
+        return [...acc, row]
+    }, [])
+    const { url: linkToFile } = await createExportFile({
+        fileName: `news_recipients_${dayjs().format('DD_MM')}.xlsx`,
+        templatePath: './domains/news/templates/NewsRecipientsExportTemplate.xlsx',
+        replaces: {
+            header: HeaderMessage,
+            newsRecipients: processedRows,
+            i18n: {
+                ...getHeadersTranslations(EXPORT_TYPE_NEWS_RECIPIENTS, locale),
+                sheetName: i18n('excelExport.sheetNames.newsRecipients', { locale }),
+            },
+        },
+    })
+
+    return linkToFile
+}
+
+const ExportNewsRecipientsService = new GQLCustomSchema('ExportNewsRecipientsService', {
+    types: [
+        {
+            access: true,
+            type: 'input ExportNewsRecipientsInput { dv: Int!, sender: JSON!, organizationId: ID!, newsItemScopes: [NewsItemScopeWhereInput] }',
+        },
+        {
+            access: true,
+            type: 'type ExportNewsRecipientsOutput { status: String!, linkToFile: String! }',
+        },
+    ],
+    
+    mutations: [
+        {
+            access: access.canExportNewsRecipients,
+            schema: 'exportNewsRecipients(data: ExportNewsRecipientsInput!): ExportNewsRecipientsOutput',
+            resolver: async (parent, args, context, info, extra = {}) => {
+                const { data: { newsItemScopes, organizationId } } = args
+                const locale = extractReqLocale(context.req) || conf.DEFAULT_LOCALE
+
+                let propertyIds = []
+                if (!isEmpty(newsItemScopes)) {
+                    propertyIds = compact(map(newsItemScopes, ['property', 'id']))
+                }
+
+                let residentsByProperties = []
+                if (!isEmpty(propertyIds)) {
+                    residentsByProperties = await loadListByChunks({
+                        context,
+                        list: Resident,
+                        chunkSize: 50,
+                        where: {
+                            property: {
+                                id_in: propertyIds,
+                            },
+                            deletedAt: null,
+                        },
+                    })
+                }
+
+                const recipientsByOrganization = []
+                if (organizationId && isEmpty(newsItemScopes)) {
```

Comment:
Redundant check of `organizationId` because it is a required variable in GraphQL input
```suggestion
                if (isEmpty(newsItemScopes)) {
```